nemo/collections/nlp/models/language_modeling/megatron_gpt_model.py

```diff
@@ -534,6 +539,33 @@ def initialize_ub_func(self):
         )
         self.initialize_ub = False
 
+    def on_train_batch_start(self, batch, batch_idx):
+        super().on_train_batch_start(batch, batch_idx)
+        P = parallel_state.get_pipeline_model_parallel_rank()
+        T = parallel_state.get_tensor_model_parallel_rank()
+        D = parallel_state.get_data_parallel_rank()
+        R = torch.distributed.get_rank()
+
+        self._nvtx_range = nvtx.start_range(f'Training Step [GPU rank {R} (D{D}P{P}T{T})]', color='green')
+
+        self.timestamp = None
+        if R == 0:
+            self.timestamp = time.time_ns()
+
+    def on_train_batch_end(self, outputs, batch, batch_idx):
+        super().on_train_batch_end(outputs, batch, batch_idx)
+        nvtx.end_range(self._nvtx_range)
+
+        if self.timestamp:
+            now = time.time_ns()
+            self.log_step_time(now - self.timestamp)
+            self.timestamp = None
+
+    def log_step_time(self, step_time_ns):
+        step_time = step_time_ns / 1000000000
+        instance_zone = "unknown"
```

Comment:
what is `instance_zone` for?

Reply:
Deleted, an artifact from the original patch code I got from Patrick.